### ISSUE TITLE
No nullable warning issued when enumerating a Nullable structure

### ISSUE THREAD

**Issue by jasonmalinowski** [dotnet/roslyn#43772]
```
#nullable enable

class Program
{
    static async Task Main(string[] args)
    {
        foreach (var x in await GetPotentiallyNullList())
        {
        }
    }

    static ValueTask<ImmutableArray<string>?> GetPotentiallyNullList()
    {
        return new ValueTask<ImmutableArray<string>?>(result: null);
    }
}
```

This is apparently expected that it builds, due to a conversation discussed at length [here](https://twitter.com/samharwell/status/539796073946963968). However, I would have expected that this should still raise a nullable warning due to the implicit access of the .Value property in the foreach.

**Comment by jcouv** [comment]
Agreed.
The problem also repro's without `foreach`: `(await GetPotentiallyNullList()).Value.ToString();` (doesn't produce a warning, but should)

Tagging @cston

FWIW, removing `await`/`Task` yields the proper warning:
``` csharp
#nullable enable
using System.Collections.Immutable;

class Program
{
    static void Main(string[] args)
    {
        foreach (var x in  GetPotentiallyNullList()) // warning CS8629: Nullable value type may be null.
        {
        }
    }

    static ImmutableArray<string>? GetPotentiallyNullList()
    {
        throw null!;
    }
}
```
